tests/test_minimizer.py

```diff
@@ -0,0 +1,27 @@
+
+import numpy as np
+
+from rdkit import Chem
+
+from md import minimizer, builders
+
+from ff.handlers.deserialize import deserialize_handlers
+from ff import Forcefield
+
+def test_minimizer():
+
+    complex_system, complex_coords, _, _, complex_box, _ = builders.build_protein_system('tests/data/hif2a_nowater_min.pdb')
+    complex_box += np.eye(3)*0.1 # BFGS this later
+
+    suppl = Chem.SDMolSupplier('tests/data/ligands_40.sdf', removeHs=False)
+    all_mols = [x for x in suppl]
+    mol_a = all_mols[1]
+    mol_b = all_mols[4]
+
+
+    ff = Forcefield(deserialize_handlers(open('ff/params/smirnoff_1_1_0_ccc.py').read()))
+
+    # these methods will throw if the minimization failed
+    minimized_coords = minimizer.minimize_host_4d([mol_a, mol_b], complex_system, complex_coords, ff, complex_box)
+    minimized_coords = minimizer.minimize_host_4d([mol_a], complex_system, complex_coords, ff, complex_box)
+    minimized_coords = minimizer.minimize_host_4d([mol_b], complex_system, complex_coords, ff, complex_box)
```

Comment:
Minor possible improvement: We could have each of these three calls act as a separate pytest, using pytest.mark.parameterize on the first argument.

Reply:
Cool! I need to look up the documentation on this, didn't know about it

Reply:
okay I actually tried this, but for this particular use case it's a little tricky since loading in the values as an arg to parameterize was difficult, and using idxs requires too much janking around broadcasting (since python lists aren't broadcastable). I think it's probably clear/easier to read as-is.

```

import numpy as np

from rdkit import Chem

from md import minimizer, builders

from ff.handlers.deserialize import deserialize_handlers
from ff import Forcefield

import pytest

@pytest.mark.parametrize("mol_idxs", [[0, 1], [0], [1]])
def test_minimizer(mol_idxs):

    complex_system, complex_coords, _, _, complex_box, _ = builders.build_protein_system('tests/data/hif2a_nowater_min.pdb')
    complex_box += np.eye(3)*0.1 # BFGS this later

    suppl = Chem.SDMolSupplier('tests/data/ligands_40.sdf', removeHs=False)
    all_mols = [x for x in suppl]
    # mol_a = all_mols[1]
    # mol_b = all_mols[4]


    ff = Forcefield(deserialize_handlers(open('ff/params/smirnoff_1_1_0_ccc.py').read()))

    # these methods will throw if the minimization failed
    # can't easily broadcast this
    minimized_coords = minimizer.minimize_host_4d(all_mols[mol_idxs], complex_system, complex_coords, ff, complex_box)
`````